src/Plugin/Action/BlockIP.php

```diff
@@ -0,0 +1,88 @@
+<?php
+
+/**
+ * @file
+ * Contains \Drupal\rules\Plugin\Action\BlockIP.
+ */
+
+namespace Drupal\rules\Plugin\Action;
+
+use Drupal\ban\BanIpManagerInterface;
+use Drupal\Component\Utility\String;
+use Drupal\Core\Plugin\ContainerFactoryPluginInterface;
+use Drupal\rules\Engine\RulesActionBase;
+use Symfony\Component\DependencyInjection\ContainerInterface;
+
+/**
+ * Provides the 'Block IP' action.
+ *
+ * @Action(
+ *   id = "rules_block_ip",
+ *   label = @Translation("Block IP"),
+ *   category = @Translation("System"),
+ *   context = {
+ *     "ip" = @ContextDefinition("string",
+ *       label = @Translation("IP Address")
+ *     )
+ *   }
+ * )
+ *
+ * @todo: Add access callback information from Drupal 7.
+ * @todo: We should maybe use a dedicated data type for the ip address, as we
+ * do in Drupal 7.
+ * @todo: This action depends on the ban module. We need to have a way to
+ * specify this.
+ */
+class BlockIP extends RulesActionBase implements ContainerFactoryPluginInterface {
+
+  /**
+   * The ban manager service used to ban the IP.
```

Comment:
Minor, but the class does not know it's a service. So it's just The ban manager.